src/test/java/Swagger/lesson3/__swagger/controller/StudentControllerWebMvcTest.java

```diff
@@ -0,0 +1,154 @@
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .put("/student")
+                        .content(new ObjectMapper().writeValueAsString(new Student(1L,"Анна Борисова", 13)))
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id").value(expected.getId()))
+                .andExpect(jsonPath("$.name").value(expected.getName()))
+                .andExpect(jsonPath("$.age").value(expected.getAge()));
+    }
+
+    @Test
+    void getAll() throws Exception {
+        List<Student> students = List.of(
+                new Student("Сергей Иванов", 11),
+                new Student("Роман Григорьев", 12),
+                new Student("Ольга Сергеева", 13),
+                new Student("Андрей Смирнов", 14),
+                new Student("Ксения Захарова", 15));
+        when(studentService.getAll()).thenReturn(students);
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/student")
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0]").value(students.get(0)))
+                .andExpect(jsonPath("$[1]").value(students.get(1)))
+                .andExpect(jsonPath("$[2]").value(students.get(2)))
+                .andExpect(jsonPath("$[3]").value(students.get(3)))
+                .andExpect(jsonPath("$[4]").value(students.get(4)));
+    }
+
+    @Test
+    void getByAgeBetweenTest() throws Exception {
+        int minAge = 12;
+        int maxAge = 15;
+        List<Student> students = List.of(
+                new Student("Сергей Иванов", 11),
+                new Student("Роман Григорьев", 12),
+                new Student("Ольга Сергеева", 13),
+                new Student("Андрей Смирнов", 14),
+                new Student("Ксения Захарова", 15));
+        when(studentService.getByAgeBetween(any(Integer.class), any(Integer.class))).thenReturn(students);
+
+        mockMvc.perform(MockMvcRequestBuilders
+                        .get("/student/getByAgeBetween?minAge=12&maxAge=15")
+                        .param("minAge", String.valueOf(minAge))
+                        .param("maxAge", String.valueOf(maxAge))
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0]").value(students.get(0)))
+                .andExpect(jsonPath("$[1]").value(students.get(1)));
+    }
+
+    @Test
+    void getFacultyOfStudent() throws Exception {
+        Student student = new Student(1L, "Анна Борисова", 13);
+        Faculty faculty = new Faculty(1L, "История", "Синий");
+
+        student.setFaculty(faculty);
+        when(studentRepository.findById(1L)).thenReturn(Optional.of(student));
```

Comment:
Сервис это мок, его поведение и нужно конфигурировать. До репозиториев код даже не доходит
```suggestion
        when(studentService.get(1L)).thenReturn(student);
```